src/formatFiles.js

```diff
@@ -1,27 +1,49 @@
-import { readFileSync, writeFileSync } from 'fs';
-import { resolveConfig, format } from 'prettier';
+import { existsSync, readFileSync, writeFileSync } from 'fs';
+import { resolveConfig, getFileInfo, format as prettierFormat } from 'prettier';
 import { join } from 'path';
 
 export default (
-  directory,
+  rootDirectory,
   files,
   { config, onWriteFile, onExamineFile } = {}
 ) => {
+  // Use local "prettier-tslint" if possible.
+  const tslintFormat = loadFormat(rootDirectory, 'prettier-tslint', 'format');
+  // Use local "prettier-eslint" if possible.
+  const eslintFormat = loadFormat(rootDirectory, 'prettier-eslint');
+
   for (const relative of files) {
     onExamineFile && onExamineFile(relative);
-    const file = join(directory, relative);
-    const options = resolveConfig.sync(file, { config, editorconfig: true });
+    const file = join(rootDirectory, relative);
     const input = readFileSync(file, 'utf8');
-    const output = format(
-      input,
-      Object.assign({}, options, {
-        filepath: file,
-      })
-    );
+
+    const parser = getFileInfo.sync(file).inferredParser;
+    const format =
+      (parser === 'babylon' && (eslintFormat || tslintFormat)) ||
```

Comment:
Or `flow`

Reply:
Does eslint support flow?

Reply:
`babylon` supports flow language too. You can write JavaScript and still use the `flow` parser.